### PR TITLE
chore(jkube-kit-build-api): Removed org.mockito.Mockito references fr…

### LINKED ISSUE #2657: Remove org.mockito.Mockito. references from AssemblyConfigurationUtilsTest
### Component

JKube Kit

### Description
Related to https://github.com/eclipse/jkube/issues/2316

AssemblyConfigurationUtilsTest seems to be using mock at some places, which can be refactor to use real objects.

### Expected Behavior
Usage of org.mockito.Mockito.mock is removed from AssemblyConfigurationUtilsTest

### Acceptance Criteria
- [ ] mock statements are replaced via real variables using builders
- [ ] AssemblyConfigurationUtilsTest should be passing even after refactoring the test

### PR DESCRIPTION
**Pull request by ShantKhatri**
…om AssemblyConfigurationUtilsTest 

## Description

Fixes #2657 
Replaced org.mockito.Mockito references from AssemblyConfigurationUtilsTest  with real objects and builders

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2658 (_2024-02-12T11:26:06Z_)

:gear: [JKube E2E Tests (7871022643)](https://github.com/jkubeio/ci/actions/runs/7871022643) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2658)](https://github.com/jkubeio/ci/actions/runs/7871022643/job/21473431349)

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java [hunk 1]
```diff
@@ -56,11 +62,14 @@ void getAssemblyConfigurationOrCreateDefault_withNoConfiguration_shouldReturnDef
   @Test
   void getAssemblyConfigurationOrCreateDefault_withConfiguration_shouldReturnConfiguration() {
     // Given
-    BuildConfiguration mockedBuildConfiguration = mock(BuildConfiguration.class);
-    final AssemblyConfiguration configuration = AssemblyConfiguration.builder().user("OtherUser").name("ImageName").build();
-    when(mockedBuildConfiguration.getAssembly()).thenReturn(configuration);
+    BuildConfiguration buildConfig = BuildConfiguration.builder()
```
Comment (rohanKanojia) [inline jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java:65]: why are you creating a local variable of the same name as class level `buildConfig` here?

Comment (ShantKhatri) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java:65]: Actually test _getAssemblyConfigurationOrCreateDefault_withConfiguration_shouldReturnConfiguration_ is fails when I define buildConfig here like this:
`
buildConfig.toBuilder()
            .assembly(AssemblyConfiguration.builder()
                    .user("OtherUser")
                    .name("ImageName")
                    .build())
            .build();
`
after declaring it as above, I'm getting assembly configurations as null.
That's why I have redeclared it again.

Comment (rohanKanojia) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java:65]: Does it work if you reassign `buildConfig` variable to new value?

Comment (rohanKanojia) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java:65]: Could you please replace it by directly reassigning `buildConfig`?

Comment (ShantKhatri) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java:65]: Sorry, I was confused with terms.
It not works as expected when I reassign the new value to `buildConfig`.
When I reassign `buildConfig` variable to new value I'm getting `buildConfig.getAssembly()` as `null`.
and when I declare it as  
```
BuildConfiguration buildConfig = BuildConfiguration.builder()
            .assembly(AssemblyConfiguration.builder()
                    .user("OtherUser")
                    .name("ImageName")
                    .build())
            .build();
```

it works as expected.
I don't know why this things happens.

Comment (rohanKanojia) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java:65]: I'm asking you to modify it like this:

```suggestion
    buildConfig = buildConfig.toBuilder()
```

Comment (ShantKhatri) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java:65]: Sure, Thanks! It works.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**